common/protocol_parser/src/es_log/parser.rs

```diff
@@ -0,0 +1,227 @@
+use std::borrow::Cow;
+
+use models::utils::now_timestamp_millis;
+use protos::FieldValue;
+use serde::{Deserialize, Serialize};
+use serde_json;
+use snafu::Snafu;
+
+use crate::Line;
+
+const ES_LOG_MSG_FIELD: &str = "_msg";
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Fields {
+    #[serde(rename = "_msg")]
+    msg: Option<String>,
+    #[serde(rename = "_time")]
+    pub time: Option<String>,
+    #[serde(flatten)]
+    ohters: serde_json::Map<String, serde_json::Value>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Command {
+    #[serde(rename = "create")]
+    Create(CommandInfo),
+    #[serde(rename = "index")]
+    Index(CommandInfo),
+    Invalid,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct CommandInfo {
+    pub _id: Option<String>,
+    pub _index: Option<String>,
+}
+
+pub struct ESLog {
+    pub command: Command,
+    pub fields: Fields,
+}
+
+pub struct Parser {
+    table: String,
+    // TODO Some statistics here
+}
+
+#[derive(Debug, Snafu, PartialEq, Eq)]
+pub enum Error {
+    #[snafu(display("invalid es log syntax"))]
+    InvaildSyntax,
+
+    #[snafu(display("Common: '{}'", content))]
+    Common { content: String },
+
+    #[snafu(display("invalid time format"))]
+    ParseTime,
+}
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+impl Parser {
+    pub fn new(table: String) -> Self {
+        Self { table }
+    }
+
+    pub fn parse<'a>(
+        &self,
+        req: &'a str,
+        time_alias: Option<String>,
+        msg_alias: Option<String>,
+    ) -> Result<Vec<Line<'a>>> {
+        let mut ret: Vec<Line> = Vec::new();
+
+        let json_chunk: Vec<&str> = req.trim().split('\n').collect();
+        let lines_len = json_chunk.len();
+        if lines_len % 2 != 0 {
```

Comment:
It is advisable to add a comment explaining why the remainder 2 is taken, for example:
/*
* because the es log is a pair of command and fields like:
{ "index" : { "_index" : "test", "_id" : "1" } }
{ "field1" : "value1" }
{ "create" : { "_index" : "test", "_id" : "3" } }
{ "field1" : "value3" }
, so the lines must be even number
*/